subprojects/logging/src/main/java/org/gradle/util/SingleMessageLogger.java

```diff
@@ -211,6 +211,12 @@ public static void nagUserOfDeprecatedBehaviour(String behaviour) {
         }
     }
 
+    public static void nagUserOfDeprecatedThing(String thing) {
+        if (isEnabled()) {
+            nagUserOfDeprecated(String.format("%s. This", thing));
```

Comment:
:o: Maybe here use "This domain object".

Reply:
Well, we do not deprecate the domain object. `This domain object is deprecated in...` is not true. We only deprecate certain naming. But `This naming is deprecated...` Sounds weird to me. I prefer just `This is deprecated...`, where _this_ refers to the sentence before. I do not have a better idea and I think the complete composed message for the user sounds good.